server/src/main/java/io/crate/planner/optimizer/rule/MergeFilterAndCollect.java

```diff
@@ -49,6 +49,13 @@ public MergeFilterAndCollect() {
             .with(source(), typeOf(Collect.class).capturedAs(collectCapture));
     }
 
+    @Override
+    public boolean mandatory() {
+        // The rule is mandatory, some queries will fail in the execution phase if excluded
+        // and many tests rely on the output of the logical plan on this
```

Comment:
I'm sorry but this reads a lot like "It's mandatory because it's mandatory".

Why do some queries fail in the execution phase?

If tests rely on the output of the logical plan, we should use the rule annotation for these and not make the rule in general mandatory.

Reply:
I haven't debugged the execution phase, but you are right I can annotate where the output does not fit.

Reply:
I will do this as a follow-up, this is affecting over 100 tests.